src/main/src/main/java/com/shevtsov/servlets/WebApp.java

```diff
@@ -19,14 +24,20 @@
 
     @Override
     public void contextInitialized(ServletContextEvent sce) {
-        ClientDao clientDao = new ClientDBDao(new DBConnectionWorkDB());
+        DBConnection dbConnection = new DBConnectionWorkDB();
+        ClientDao clientDao = new ClientDBDao(dbConnection);
         ValidationService validationService = new ValidationServiceImpl(clientDao);
         AuthorisationImpl authorisation = new AuthorisationImpl(clientDao);
         ClientService clientService = new ClientServiceImpl(validationService, clientDao, authorisation);
+        ProductDao productDao = new ProductDBDao(dbConnection);
+        ProductService productService = new ProductServiceImpl(productDao);
+
 
         ServletContext servletContext = sce.getServletContext();
         servletContext.addServlet("ClientServlet", new ClientServlet(clientService)).
                 addMapping("/clients/*");
+        servletContext.addServlet("ProductServlet", new ProductServlet(productService)).
```

Comment:
Why do not use web.xml

Reply:
Artem showed us 3 ways: web.xml, annotations and ServletContextListener implementation. It was decided to use the last one to create all objects in one place and to use dependency injection.

Reply:
I prefer web.xml with listener